From 059512b7c8e95f6b4e5bde6dca62523b769f8cd7 Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Mon, 30 Nov 2020 08:42:25 -0800
Subject: [PATCH] Add "own" crate back to list of dependent crates

Commit 75a2f5b83302b7b747cfb41e75d4d9b2adfff92f caused the program
itself to be added to the list of dependencies, which in turn meant that
Portage will not fetch that package and, hence, the source is not
available for compilation.
This change reverts said commit to preserve the desired behavior.
---
 src/lib.rs | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/lib.rs b/src/lib.rs
index 637846..0b1ec4 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -108,11 +108,7 @@ pub fn gen_ebuild_data(manifest_path: Option<PathBuf>) -> Result<EbuildConfig> {
     let lockfile = Lockfile::load(lockfile_path)?;
 
     for pkg in lockfile.packages {
-        if let Some(src) = pkg.source {
-            if src.is_default_registry() {
-                crates.push(format!("\t{}-{}\n", pkg.name, pkg.version));
-            }
-        }
+        crates.push(format!("\t{}-{}\n", pkg.name, pkg.version));
     }
 
     Ok(EbuildConfig::from_package(root_pkg, crates, licenses))
-- 
2.32.0

